File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        n=0
        sign=-1 if x<0 else 1
        x=abs(x)
        while(x):
            r=x%10
            if(n>214748364 or n<-214748364):
                return 0
            n=n*10+r
            x//=10
        
        return sign*n

        
        
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int sign = 1;
        if ((long long)x < INT_MIN || (long long)x > INT_MAX)
            return 0;
        if (x < 0) {
            sign = -1;
            if ((long long)x * -1 > INT_MAX)
                return 0;
            x *= -1;
        }

        int newnum = 0;
        while (x > 0) {
            int rem = x % 10;
            long long temp = (long long)newnum * 10 + rem;
            if (temp > INT_MAX) {
                return 0;
            }

            newnum = temp;
            x /= 10;
        }

        return newnum * sign;
    }
};
